src/client/components/app/SpigotDisplay/SpigotMetadata.tsx

```diff
@@ -0,0 +1,261 @@
+  const { NETWORK } = getEnv();
+  const connectWallet = () => dispatch(WalletActions.walletSelect({ network: NETWORK }));
+  const network = useAppSelector(NetworkSelectors.selectCurrentNetwork);
+  const etherscanUrl = getEtherscanUrlStub(network);
+  const tokensMap = useAppSelector(TokensSelectors.selectTokensMap);
+
+  useEffect(() => {
+    if (!selectedSpigot || !selectedSpigot.spigots) return;
+    const spigots = selectedSpigot.spigots ?? {};
+    const spigotIds = Object.keys(selectedSpigot.spigots!) ?? [];
+    for (let i = 0; i < spigotIds.length; i++) {
+      const id = spigotIds[i];
+      const revenueContract = spigots[id].contract;
+      dispatch(OnchainMetaDataActions.getABI(revenueContract));
+    }
+  }, [selectedSpigot]);
+
+  // if (!selectedLine) return <Container>{t('lineDetails:line.no-data')}</Container>;
+  if (!selectedSpigot) return <Container>{t('lineDetails:line.no-data')}</Container>;
+  // const { spigot } = selectedLine;
+
+  const addSpigotHandler = (token: TokenView) => {
+    if (!walletIsConnected) {
+      connectWallet();
+    } else {
+      dispatch(CollateralActions.setSelectedRevenueContract({ contractAddress: token.address }));
+      dispatch(ModalsActions.openModal({ modalName: 'enableSpigot' }));
+    }
+  };
+
+  const enableSpigotText = walletIsConnected
+    ? `${t('spigot:metadata.add-revenue-contract')}`
+    : `${t('components.connect-button.connect')}`;
+
+  const getCollateralTableActions = () => {
+    return <Button onClick={addSpigotHandler}>{enableSpigotText}</Button>;
+  };
+
+  const claimRev = (contract: string) => {
+    return {
+      name: t('spigot:claim-revenue'),
+      handler: () => {
+        dispatch(
+          ModalsActions.openModal({ modalName: 'claimRevenue', modalProps: { revenueContractAddress: contract } })
+        );
+        // open modal, dispatch action inside modal
+        // dispatch(
+        //   CollateralActions.claimRevenue({
+        //     spigotAddress: selectedSpigot.id,
+        //     revenueContract: selectedRevenueContract
+        //   })
+        // )
+      },
+    };
+  };
+
+  const allSpigots: SpigotRevenueContract[] = [...Object.values(selectedSpigot?.spigots ?? {})];
+  const formattedSpigots = allSpigots.map((c) => ({
+    ...c,
+    key: c.contract,
+    // header: c.type + c.token.toString(),
+    align: 'flex-start',
+    actions: 'claim-revenue',
+  }));
+  console.log('formatted spigots: ', formattedSpigots);
+  return (
+    <>
+      <ViewContainer>
+        <AssetsListCard
+          header={' '} //{t('lineDetails:metadata.escrow.assets-list.title')}
+          data-testid="line-assets-list"
+          metadata={[
+            {
+              key: 'contract',
+              header: 'Contract Address', //t('lineDetails:metadata.escrow.assets-list.contract'),
+              transform: ({ contract }) => (
+                <a href={etherscanUrl + `${contract}`} target={'_blank'} rel={'noreferrer'}>
```

Comment:
ALWAYS use components/common/Link for any internal or external navigation

Reply:
fixed